cupy/indexing/indexing.py

```diff
@@ -104,4 +104,38 @@ def diagonal(a, offset=0, axis1=0, axis2=1):
     return a.diagonal(offset, axis1, axis2)
 
 
+def extract(condition, a):
+    """Return the elements of an array that satisfy some condition.
+
+    This is equivalent to ``np.compress(ravel(condition), ravel(arr))``.
+    If ``condition`` is boolean, ``np.extract`` is equivalent to
+    ``arr[condition]``.
+
+    Args:
+        condition (cupy.ndarray): An array whose nonzero or True entries
+            indicate the elements of array to extract.
+        a (cupy.ndarray): Input array of the same size as condition.
+
+    Returns:
+        cupy.ndarray: Rank 1 array of values from arr where condition is True.
+
+    .. warning::
+
+            This function may synchronize the device.
+
+    .. seealso:: :func:`numpy.extract`
+    """
+
+    if not isinstance(a, cupy.ndarray):
+        raise TypeError('extract requires input array to be cupy.ndarray')
+
+    if not isinstance(condition, cupy.ndarray):
+        condition = cupy.array(condition)
```

Comment:
The above docstring says that the argument `condition` is required to be of `cupy.ndarray`.

Reply:
Well NumPy also allows this.

```python
In [1]: import numpy                                                            

In [2]: a = numpy.arange(6).reshape(2, 3)                                       

In [3]: a                                                                       
Out[3]: 
array([[0, 1, 2],
       [3, 4, 5]])

In [4]: numpy.extract([[True, False, True], 
   ...:                [False, True, True]], 
   ...:               a)                                                        
Out[4]: array([0, 2, 4, 5])
```

Though perhaps the question is how do we specify something that is `ndarray`-like in a way that is compatible with CuPy's documentation practices 🙂

Reply:
I too had the same doubt. There was no mention of `array_like` in CuPy's documentations.